torch/csrc/jit/serialization/export.cpp

```diff
@@ -1248,13 +1249,18 @@ std::string serialize_model_proto_to_string(
   return model_proto->SerializeAsString();
 }
 
-void check_onnx_proto(const std::string& proto_string) {
+void check_onnx_proto(const std::string& proto_string, bool full_check) {
   onnx::ModelProto model;
   if (!ParseProtoFromBytes(&model, proto_string.c_str(), proto_string.size())) {
     throw std::runtime_error("Invalid ONNX proto string.");
     return;
   }
   onnx::checker::check_model(model);
+
+  if (full_check) {
+    onnx::shape_inference::InferShapes(model);
```

Comment:
I think this needs to happen before L1258 check_model.

Reply:
In onnx repo there is a python api `check_model` which does the c++ `check_check` followed a shape inference call, in this order

But I can change the order if you think it might be more useful

Reply:
my bad, I thought it was the other way around. This is good. Thanks for pointing out.